packages/@vuepress/core/lib/dev.js

```diff
@@ -105,52 +100,49 @@ module.exports = async function dev (sourceDir, cliOptions = {}) {
     config = applyUserWebpackConfig(userConfig, config, false /* isServer */)
   }
 
+  const serverConfig = {
+    disableHostCheck: true,
+    compress: true,
+    clientLogLevel: 'none',
+    hot: true,
+    quiet: true,
+    headers: {
+      'access-control-allow-origin': '*'
+    },
+    publicPath: ctx.base,
+    watchOptions: {
+      ignored: /node_modules/
+    },
+    historyApiFallback: {
+      rewrites: [
+        { from: /\.html$/, to: '/' }
+      ]
+    },
+    overlay: false,
+    host,
+    contentBase: path.resolve(sourceDir, '.vuepress/public')
+    // before(app) {
+    //   ctx.pluginAPI.options.enhanceDevServer.syncApply(app)
+    //   app.use('/', express.static(path.resolve(sourceDir, '.vuepress/public')))
+    // }
+  }
+
+  WebpackDevServer.addDevServerEntrypoints(config, serverConfig)
+
   const compiler = webpack(config)
+  const server = new WebpackDevServer(compiler, serverConfig)
 
-  const nonExistentDir = path.resolve(__dirname, 'non-existent')
-  await serve({
-    // avoid project cwd from being served. Otherwise if the user has index.html
-    // in cwd it would break the server
-    content: [nonExistentDir],
-    compiler,
-    host,
-    dev: { logLevel: 'warn' },
-    hot: {
-      port: port + 1,
-      logLevel: 'error'
-    },
-    logLevel: 'error',
-    port,
-    open: cliOptions.open,
-    add: app => {
-      // apply plugin options to extend dev server.
-      ctx.pluginAPI.options.enhanceDevServer.syncApply(app)
-
-      const userPublic = path.resolve(sourceDir, '.vuepress/public')
-
-      // enable range request
-      app.use(range)
```

Comment:
✅Express enable range request in development. so removing it will not break the fix of https://github.com/vuejs/vuepress/pull/555.